staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -468,7 +468,7 @@ func (w *watchCache) GetAllEventsSinceThreadUnsafe(resourceVersion uint64) ([]*w
 		return result, nil
 	}
 	if resourceVersion < oldest-1 {
-		return nil, errors.NewGone(fmt.Sprintf("too old resource version: %d (%d)", resourceVersion, oldest-1))
+		return nil, errors.NewResourceExpired(fmt.Sprintf("too old resource version: %d (%d)", resourceVersion, oldest-1))
```

Comment:
is this a nice-to-have, or does the client implementation depend on this?

Reply:
This is a nice to have. @smarterclayton had noticed that everywhere else we return resource expired except for there (get,list,watch for watch cache both enabled and disabled). The HTTP Status code is 410 for both responses.

Reply:
I still check for both on the client side, of course.